src/sqlite_db.cpp

```diff
@@ -50,6 +50,243 @@
 
 namespace miopen {
 
+class SQLite::impl
+{
+    struct SQLiteCloser
+    {
+        void operator()(sqlite3* ptr)
+        {
+            std::string filename_(sqlite3_db_filename(ptr, "main"));
+            SQLite::impl::Retry([&]() { return sqlite3_close(ptr); }, filename_);
+        }
+    };
+
+    public:
+    impl(const std::string& filename_, bool is_system)
+    {
+        sqlite3* ptr_tmp;
+        int rc = 0;
+        if(is_system)
+            rc = sqlite3_open_v2(filename_.c_str(), &ptr_tmp, SQLITE_OPEN_READONLY, nullptr);
+        else
+            rc = sqlite3_open_v2(
+                filename_.c_str(), &ptr_tmp, SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE, nullptr);
+        ptrDb   = sqlite3_ptr{ptr_tmp};
+        isValid = (rc == 0);
+    }
+    bool Exec(const std::string& query) const
+    {
+        MIOPEN_LOG_T(std::this_thread::get_id() << ":" << query);
+        auto rc = Retry([&]() {
+            return sqlite3_exec(ptrDb.get(), query.c_str(), find_callback, nullptr, nullptr);
+        });
+        if(rc != SQLITE_OK)
+        {
+            MIOPEN_LOG_I2(query);
+            MIOPEN_THROW(miopenStatusInternalError, ErrorMessage());
+            sqlite3_close(ptrDb.get());
+            return false;
+        }
+        return true;
```

Comment:
Since this always returns `true` this should just be removed.

Reply:
Fixed